orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/config/ServiceSelector.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.config;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+interface ServiceSelector {
+  Object getService();
+  boolean supports(SelectableService.Criteria criteria);
+}
+
+class ByApplicationServiceSelector<T> implements ServiceSelector {
+  private final Object service;
+  private final Pattern applicationPattern;
+
+  ByApplicationServiceSelector(T service, Map<String, Object> config) {
+    this.service = service;
+    this.applicationPattern = Pattern.compile((String) config.get("applicationPattern"));
+  }
+
+  @Override
+  public Object getService() {
+    return service;
+  }
+
+  @Override
+  public boolean supports(SelectableService.Criteria criteria) {
+    if (criteria.getApplication() == null) {
+      return false;
+    }
+
+    return applicationPattern.matcher(criteria.getApplication().toLowerCase()).matches();
+  }
+}
+
+class ByExecutionTypeServiceSelector implements ServiceSelector {
+  private final Object service;
+  private final Set<String> executionTypes;
+
+  ByExecutionTypeServiceSelector(Object service, Map<String, Object> config) {
+    this.service = service;
+    this.executionTypes = new HashSet(
+      ((Map<String, String>) config.get("executionTypes")).values()
```

Comment:
I think this needs to be either:

`new HashSet(((Map<String, Collection<String>>) config).get("executionTypes"))`
or
`new HashSet((Collection<String>) config.get("executionTypes"))`
?